ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { InternalAccount, isEvmAccountType } from '@metamask/keyring-api';
+import { NetworkConfiguration } from '@metamask/network-controller';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  getInternalAccounts,
+  getNetworkConfigurationsByChainId,
+  getSelectedInternalAccount,
+  getUpdatedAndSortedAccounts,
+} from '../../../selectors';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+  Text,
+} from '../../../components/component-library';
+import {
+  Content,
+  Footer,
+  Header,
+  Page,
+} from '../../../components/multichain/pages/page';
+import { SiteCell } from '../../../components/multichain/pages/review-permissions-page';
+import {
+  BlockSize,
+  Display,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { MergedInternalAccount } from '../../../selectors/selectors.types';
+import { mergeAccounts } from '../../../components/multichain/account-list-menu/account-list-menu';
+import { TEST_CHAINS } from '../../../../shared/constants/network';
+
+export type ConnectPageRequest = {
+  id: string;
+  origin: string;
+};
+
+type ConnectPageProps = {
+  request: ConnectPageRequest;
+  permissionsRequestId: string;
+  rejectPermissionsRequest: (id: string) => void;
+  approveConnection: (request: ConnectPageRequest) => void;
+  activeTabOrigin: string;
+};
+
+export const ConnectPage: React.FC<ConnectPageProps> = ({
+  request,
+  permissionsRequestId,
+  rejectPermissionsRequest,
+  approveConnection,
+  activeTabOrigin,
+}) => {
+  const t = useI18nContext();
+
+  const networkConfigurations = useSelector(getNetworkConfigurationsByChainId);
+  const [nonTestNetworks, testNetworks] = useMemo(
+    () =>
+      Object.entries(networkConfigurations).reduce(
+        ([nonTestNetworksList, testNetworksList], [chainId, network]) => {
+          const isTest = (TEST_CHAINS as string[]).includes(chainId);
+          (isTest ? testNetworksList : nonTestNetworksList).push(network);
+          return [nonTestNetworksList, testNetworksList];
+        },
+        [[] as NetworkConfiguration[], [] as NetworkConfiguration[]],
+      ),
+    [networkConfigurations],
+  );
```

Comment:
Nit: Please move this to maybe a selector or somewhere outside of this UI component if possible.